bpf/kmesh/workload/sockops_tuple.c

```diff
@@ -23,194 +23,180 @@
+    // remote_port is network byteorder
+    // openEuler 2303 convert remote port different than other linux vendor
 #if !OE_23_03
-	tuple_key->ipv4.sport = skops->remote_port >> FORMAT_IP_LENGTH;
+    tuple_key->ipv4.sport = skops->remote_port >> FORMAT_IP_LENGTH;
 #else
-	tuple_key->ipv4.sport = skops->remote_port;
+    tuple_key->ipv4.sport = skops->remote_port;
 #endif
-	// local_port is host byteorder
-	tuple_key->ipv4.dport = bpf_htonl(skops->local_port) >> FORMAT_IP_LENGTH;
+    // local_port is host byteorder
+    tuple_key->ipv4.dport = bpf_htonl(skops->local_port) >> FORMAT_IP_LENGTH;
 }
 
 // clean map_of_auth
 static inline void clean_auth_map(struct bpf_sock_ops *skops)
 {
-	struct bpf_sock_tuple tuple_key = {0};
-	// auth run PASSIVE ESTABLISHED CB now. In thie state cb
-	// tuple info src is server info, dst is client info
-	// During the auth, src must set the client info and dst set
-	// the server info when we transmitted to the kmesh auth info.
-	// In this way, auth can be performed normally.
-	extract_skops_to_tuple_reverse(skops, &tuple_key);
-	long ret = bpf_map_delete_elem(&map_of_auth, &tuple_key);
-	if(ret && ret != -ENOENT)
-		BPF_LOG(INFO, SOCKOPS, "map_of_auth bpf_map_delete_elem failed, ret: %d\n", ret);
+    struct bpf_sock_tuple tuple_key = {0};
+    // auth run PASSIVE ESTABLISHED CB now. In thie state cb
+    // tuple info src is server info, dst is client info
+    // During the auth, src must set the client info and dst set
+    // the server info when we transmitted to the kmesh auth info.
+    // In this way, auth can be performed normally.
+    extract_skops_to_tuple_reverse(skops, &tuple_key);
+    long ret = bpf_map_delete_elem(&map_of_auth, &tuple_key);
+    if (ret && ret != -ENOENT)
+        BPF_LOG(INFO, SOCKOPS, "map_of_auth bpf_map_delete_elem failed, ret: %d\n", ret);
 }
 
 static inline void clean_dstinfo_map(struct bpf_sock_ops *skops)
 {
     __u32 *key = (__u32 *)skops->sk;
     long ret = bpf_map_delete_elem(&map_of_dst_info, &key);
-    if(ret && ret != -ENOENT)
+    if (ret && ret != -ENOENT)
         BPF_LOG(INFO, SOCKOPS, "bpf map delete destination info failed, ret: %d\n", ret);
 }
 
 // insert an IPv4 tuple into the ringbuf
 static inline void auth_ip_tuple(struct bpf_sock_ops *skops)
 {
-	struct ringbuf_msg_type *msg = bpf_ringbuf_reserve(&map_of_tuple, sizeof(*msg), 0);
-	if (!msg) {
-		BPF_LOG(WARN, SOCKOPS, "can not alloc new ringbuf in map_of_tuple");
-		return;
-	}
-	// auth run PASSIVE ESTABLISHED CB now. In thie state cb
-	// tuple info src is server info, dst is client info
-	// During the auth, src must set the client info and dst set
-	// the server info when we transmitted to the kmesh auth info.
-	// In this way, auth can be performed normally.
-	extract_skops_to_tuple_reverse(skops, &(*msg).tuple);
-	(*msg).type = (__u32)IPV4;
-	bpf_ringbuf_submit(msg, 0);
-}
-
-static inline void enable_encoding_metadata(struct bpf_sock_ops *skops)
```

Comment:
removed this function

Reply:
@bitcoffeeiux

Reply:
why removed this function

Reply:
It is unused

Reply:
The enable_encoding_metadata function adds the sock to the sockhashmap. Kernel replaces the sendmsg hook (tcp_bpf_sendmsg) of the sock and then triggers the BPF prog of the sk_msg. Otherwise, there is no hook in the normal tcp_sendmsg.